test/e2e/framework/ingress_utils.go

```diff
@@ -926,9 +927,14 @@ func (cont *GCEIngressController) backendMode(nodeports []string, keyword string
 	matchingBackendService := 0
 	for _, bs := range beList {
 		match := false
-		for _, np := range nodeports {
-			// Warning: This assumes backend service naming convention includes nodeport in the name
-			if strings.Contains(bs.Name, np) {
+		for svcName, sp := range svcPorts {
+			// Non-NEG BackendServices are named with the Nodeport in the name.
+			// NEG BackendServices' names contain the a sha256 hash of a string.
+			negString := cont.UID + cont.Ns + svcName + sp.TargetPort.String()
```

Comment:
This needs to be updated with the shorter UID and delimited fields

Reply:
Updated with shorter uid. We don't need to check for delimited fields, since we're just checking that the hash suffix matches.